test/common/router/config_impl_test.cc

```diff
@@ -134,12 +137,22 @@ Http::TestRequestHeaderMapImpl genHeaders(const std::string& host, const std::st
     hdrs.remove(":scheme");
   }
 
+  if (random_value_pair != nullptr) {
+    hdrs.setByKey(Envoy::Http::LowerCaseString(random_value_pair->first),
+                  random_value_pair->second);
+  }
   return hdrs;
 }
 
+struct OptionalGenHeadersArg {
+  std::string scheme = "http";
+  const std::pair<std::string, std::string>* random_value_pair = nullptr;
```

Comment:
I'd suggest an optional here, the copy doesn't matter in test and it seems a bit clearer semantically

Reply:
Good idea. Done.